microsoft/testsuites/dpdk/dpdktestpmd.py

```diff
@@ -264,32 +264,49 @@ def generate_testpmd_command(
             "-a --stats-period 1"
         )
 
+    def generate_testpmd_killer(self, signal: str = "KILL") -> str:
+        return f"killall -s {signal} {self.command}"
+
     def run_for_n_seconds(self, cmd: str, timeout: int) -> str:
         self._last_run_timeout = timeout
         self.node.log.info(f"{self.node.name} running: {cmd}")
-        self.timer_proc = self.node.execute_async(
-            f"sleep {timeout} && killall -s INT {cmd.split()[0]}",
+
+        # testpmd exits cleanly when killed with INT.
+        # setup some timers to kill it after timeout
+        self.node.execute_async(
+            f"sleep {timeout} && {self.generate_testpmd_killer()}",
             sudo=True,
             shell=True,
         )
         testpmd_proc = self.node.execute_async(
             cmd,
             sudo=True,
         )
-        self.timer_proc.wait_result()
         proc_result = testpmd_proc.wait_result()
         self._last_run_output = proc_result.stdout
         self.populate_performance_data()
         return proc_result.stdout
 
+    def check_testpmd_is_running(self) -> str:
+        pid_result = self.node.execute(f"pidof {self.command}")
```

Comment:
We can create a tool for `pidof` command

Reply:
We should but I don't want to in this PR. I only see one other spot using pidof so it will still be easy to swap this use over later.
We should open like, a running issue thread with tools we need to create.

Reply:
I generally try to avoid `node.execute` unless it's something very specific to the testcase. `pidof` and your logic of killing process robustly seems like something that could be used in future tests.

Reply:
I'm glad you brought this up I forgot a) there's already a kill tool and b) I added stuff to it to use here and forgot to make the second change 😂

Reply:
ok waaaay cleaner now